src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -204,8 +205,12 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
                 {
                     if (syntax != null)
                     {
-                        foreach (var loc in memberT.Locations)
+                        // PERF: Avoid accessing Locations for performance, but assert that the alternative approach is
+                        // equivalent.
+                        Debug.Assert(memberT.MergedDeclaration.Declarations.SelectAsArray(decl => decl.NameLocation).SequenceEqual(memberT.Locations));
+                        foreach (var declaration in memberT.MergedDeclaration.Declarations)
                         {
+                            var loc = declaration.NameLocation;
```

Comment:
>var loc = declaration.NameLocation; [](http://example.com/codeflow?start=28&length=35)

Is similar optimization applicable to VB?

Reply:
I wasn't able to find a similar method for VB that allows filtering by syntax tree.

Reply:
> I wasn't able to find a similar method for VB that allows filtering by syntax tree.

See methods in ```NamedTypeSymbolExtensions```: ```FindMember```, ```FindFieldOrProperty```, etc.

SemanticModel and BinderFactory often need to find a symbol from its declaration node. It is quite likely location going to be involved in this. The implementation doesn't have to be structured the same way as in C#.

Reply:
I didn't perform exhaustive search

Reply:
➡️ Filed https://github.com/dotnet/roslyn/issues/56306